client/src/sections/learning-paths/create/CourseModal.tsx

```diff
@@ -0,0 +1,175 @@
+import Modal from '@/src/shared/components/Modal/Modal';
+import SearchBar from '@/src/shared/components/SearchBar/SearchBar';
+import Table from '@/src/shared/components/Table';
+import XmarkIcon from '@/src/shared/icons/XmarkIcon';
+import React, { useEffect, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { ListItem } from '@/src/shared/components/Table/ListItem';
+import Pagination from '@/src/shared/components/Pagination';
+import Button from '@/src/shared/components/Button';
+import { useAppDispatch, useAppSelector } from '@/app/hooks';
+import { updateForm } from '@/features/learning-path/learningPathSlice';
+import { type Course } from '../../../shared/utils/interface';
+
+enum HeaderEnum {
+  Name = 'name',
+  Uploaded_By = 'uploaded_by',
+  Data_Uploaded = 'created_at',
+  Date_Modified = 'updated_at',
+}
+
+interface CourseModalProps {
+  title: string;
+  state: boolean;
+  submitButtonTitle?: string;
+  closeModal: () => void;
+}
+
+interface CourseModalForm {
+  courses: string[];
+}
+
+const CourseModal = ({
+  state,
+  submitButtonTitle = 'Submit',
+  title,
+  closeModal,
+}: CourseModalProps): JSX.Element => {
+  const { courses } = useAppSelector((state) => state.learningPath.values);
+  const [newCourses, setNewCourses] = useState<Course[]>([]);
+
+  const dispatch = useAppDispatch();
+  const { register, reset, handleSubmit } = useForm<CourseModalForm>();
+
+  useEffect(() => {
+    setNewCourses(courses);
+  }, [courses]);
+
+  const handleClose = (): void => {
+    setNewCourses(courses);
+    closeModal();
+    reset();
+  };
+
+  const handleCheckboxChange = (data: string): void => {
+    const newData: Course = JSON.parse(data);
+    if (newCourses.some((course) => course.id === newData.id)) {
+      setNewCourses(newCourses.filter((course) => course.id !== newData.id));
+    } else setNewCourses([...newCourses, newData]);
+  };
+
+  const onSubmit = ({ courses }: CourseModalForm): void => {
+    const parsedCourses = courses.map((course, index) => {
+      return { ...JSON.parse(course), order: index };
+    });
```

Comment:
The way you are assigning an order based on the index of course list loop is gonna cause bug, in fact, I believe it should be causing a bug rn, because the order of courses usually does not change on modal, unless user searches, but the point is, the index right there is based on the list of modal, so if we reorder the course from added list, and open the modal and add new one, the order of course are gonna get back to where it was before, based on the list of courses on the modal

Please revise this one